lib/colorls/flags.rb

```diff
@@ -34,6 +36,52 @@ def process
 
     private
 
+    def add_sort_options(options)
+      options.separator ''
+      options.separator 'sorting options:'
+      options.separator ''
+      options.on('-sMODE', %w[d f], 'sort (f)iles or (d)irectories first') do |m|
+        @opts[:sort] = m == 'd' ? :dirs : :files
+      end
+      options.on('--sort-dirs', '--group-directories-first', 'sort directories first') { @opts[:sort] = :dirs }
+      options.on('--sort-files', 'sort files first')                                   { @opts[:sort] = :files }
+    end
```

Comment:
I already did. But since optparse does not support multi character short options (AFAICS), I had to model that using a required parameter to the newly introduced short option `-s` (similar to what @GladOSkar suggested [here](https://github.com/athityakumar/colorls/issues/74#issuecomment-314454679)

`-sMODE                           sort (f)iles or (d)irectories first`

MODE could be 'f' or 'd' which mimics the behavior from before. Any other character raises an error.